plugins/async-node/ios/Tests/AsynNodePluginTests.swift

```diff
@@ -292,6 +284,166 @@ class AsyncNodePluginTests: XCTestCase {
+            XCTAssertNil(node)
+            return .singleNode(.concrete(JSValue()))
+        }
+
+        let asyncNodePluginPlugin = AsyncNodePluginPlugin()
+        let plugin = AsyncNodePlugin(plugins: [asyncNodePluginPlugin], resolveHandler)
+
+        plugin.context = context
+
+        XCTAssertNotNil(asyncNodePluginPlugin.context)
+
+        let player = HeadlessPlayerImpl(plugins: [ReferenceAssetsPlugin(), plugin], context: context ?? JSContext())
+
+        player.start(flow: .asyncNodeJson, completion: {_ in})
+
+        wait(for: [handlerExpectation], timeout: 5)
+    }
+
+    func testAsyncNodeWithUndefinedNode() {
+        let handlerExpectation = XCTestExpectation(description: "undefined node handled")
+
+        let context = JSContext()
+
+        let resolveHandler: AsyncHookHandler = { node in
+            handlerExpectation.fulfill()
+            XCTAssertNil(node)
+            return .singleNode(.concrete(JSValue()))
+        }
+
+        let asyncNodePluginPlugin = AsyncNodePluginPlugin()
+        let plugin = AsyncNodePlugin(plugins: [asyncNodePluginPlugin], resolveHandler)
+
+        plugin.context = context
+
+        XCTAssertNotNil(asyncNodePluginPlugin.context)
+
+        let player = HeadlessPlayerImpl(plugins: [ReferenceAssetsPlugin(), plugin], context: context ?? JSContext())
+
+        player.start(flow: .asyncNodeJson, completion: {_ in})
+
+        wait(for: [handlerExpectation], timeout: 5)
+    }
+
+        func testHandleMultipleUpdatesThroughCallbackMechanism() {
+            let handlerExpectation = XCTestExpectation(description: "first data did not change")
+
+            let context = JSContext()
+            var count = 0
+
+            let resolve: AsyncHookHandler = { _ in
+                handlerExpectation.fulfill()
+
+                if count == 1 {
+                    return .multiNode([
+                        ReplacementNode.concrete(context?.evaluateScript("""
+                            (
+                                {"asset": {"id": "text", "type": "text", "value":"1st value in the multinode"}}
+                               )
+                            """) ?? JSValue()),
+                        ReplacementNode.encodable(AsyncNode(id: "id"))])
+                } else if count == 2 {
+                    return .multiNode([
+                        ReplacementNode.encodable(AssetPlaceholderNode(asset: PlaceholderNode(id: "text-2", type: "text", value: "2nd value in the multinode"))),
+                        ReplacementNode.encodable(AsyncNode(id: "id-1"))])
+                } else if count == 3 {
+                    return .singleNode(ReplacementNode.encodable(
+                        AssetPlaceholderNode(asset: PlaceholderNode(id: "text", type: "text", value: "3rd value in the multinode"))
+                    ))
+                }
+
+                return .singleNode(ReplacementNode.concrete(context?.evaluateScript("") ?? JSValue()))
+            }
```

Comment:
This isn't using the callback mechanism, right? This looks to be just using the return value approach towards resolving the values, which would only enable a single resolution. It looks like this handler is triggered multiple times, but for explicitly different async nodes (`ReplacementNode.encodable(AsyncNode(id: "id-1"))])`.

Reply:
this needs an update